modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -44,6 +44,11 @@ validParams<MechanicalContactConstraint>()
   params.addCoupledVar("disp_x", "The x displacement");
   params.addCoupledVar("disp_y", "The y displacement");
   params.addCoupledVar("disp_z", "The z displacement");
+  params.addCoupledVar("lm", "The lagrange multiplier");
```

Comment:
This class was kind of a beast already before you added your code.  I'm reluctant to add all of this additional stuff that is specific to LM contact to the class that we use for everything else. Does it make sense to have an LM-specific class that inherits from this or something?